nexus/db-queries/src/db/datastore/deployment.rs

```diff
@@ -403,6 +495,56 @@ impl DataStore {
             zones_config.sort();
         }
 
+        // Load all the physical disks for each sled.
+        {
+            use db::schema::bp_omicron_physical_disk::dsl;
+
+            let mut paginator = Paginator::new(SQL_BATCH_SIZE);
+            while let Some(p) = paginator.next() {
+                // `paginated` implicitly orders by our `id`, which is also
+                // handy for testing: the physical disks are always consistently ordered
+                let batch = paginated(
+                    dsl::bp_omicron_physical_disk,
+                    dsl::id,
+                    &p.current_pagparams(),
+                )
+                .filter(dsl::blueprint_id.eq(blueprint_id))
+                .select(BpOmicronPhysicalDisk::as_select())
+                .load_async(&*conn)
+                .await
+                .map_err(|e| {
+                    public_error_from_diesel(e, ErrorHandler::Server)
+                })?;
+
+                paginator = p.found_batch(&batch, &|d| d.id);
+
+                for d in batch {
+                    let sled_disks = blueprint_disks
+                        .get_mut(&TypedUuid::<SledKind>::from_untyped_uuid(
+                            d.sled_id,
+                        ))
+                        .ok_or_else(|| {
+                            // This error means that we found a row in
+                            // bp_omicron_physical_disk with no associated record in
+                            // bp_sled_omicron_physical_disks.  This should be
+                            // impossible and reflects either a bug or database
+                            // corruption.
+                            Error::internal_error(&format!(
+                                "disk {:?}: unknown sled: {:?}",
+                                d.id, d.sled_id
+                            ))
+                        })?;
+                    let disk = d.into_blueprint_disk_config();
+                    sled_disks.disks.push(disk);
+                }
+            }
+        }
+
+        // Sort all zones to match what blueprint builders do.
+        for (_, disks_config) in blueprint_disks.iter_mut() {
+            disks_config.disks.sort_by(|a, b| a.id.partial_cmp(&b.id).unwrap());
```

Comment:
I think this would be equivalent and faster (allow unstable sorting, no unwrap needed):

```suggestion
            disks_config.disks.sort_unstable_by_key(|d| d.id);
```